pytools/__init__.py

```diff
@@ -678,33 +674,34 @@ def memoize_on_first_arg(function, cache_dict_name=None):
     """
 
     if cache_dict_name is None:
-        cache_dict_name = intern(
-                f"_memoize_dic_{function.__module__}{function.__name__}"
-                )
+        cache_dict_name = (memoize_on_first_arg, function)
```

Comment:
Does this work if `function` is nested (i.e. not in the global namespace)? It should keep changing its hash/id in that case, right?

Reply:
Yeah, nested functions are a difficult case, mainly because it's hard to get stable identifiers for them. My understanding is that, at least, it would fail "safe" (separate caches for each function) using this code.

IMO, `@memoize_in` is more appropriate for nested functions.

Reply:
Maybe add a small note to the docs that it doesn't work for nested?

Reply:
Besides that, it looks good to me! :+1:

Reply:
219e111